### PR TITLE
PCA index

### PR DESCRIPTION
**Pull request by amrita-kar**
# Title: PCA plot removing the Index dependence 
**Problem:** The PCA plot fails when there is a interchange between Index and SampleID.

**Solution:** I changed the npxProcessing.R function to group_by only SampleID and not Index field anymore. To check this I randomized the Index field to contain SampleIDs instead of Index. As a result, even if there is an interchange between SampleID and Index the npx processing can happen with SampleIDs only.

**Key Features:**

* The above changes were tested on the PCA plot only for now comparing the old PCA minimal PCA plot with the PCA plot with index dependence removed 

## Checklist
- [x] Make sure you are requesting to **pull a feature/bugfix branch** (right side). This should not be main or develop.
- [x] Make sure you are make a pull request against either **main or develop** (left side). (Requesting to main should be reserved for bugfixes and new releases)
- [x] Add or update unit tests (if applicable)
- [x] Check your code with any unit tests (Run devtools::check() locally)
- [ ] Add neccessary documentation (if applicable)

## Type of changes

What type of changes does your code introduce?

- [ ] Bugfix (non-breaking change which fixes an issue) (link the issue on the right)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code style update (formatting, renaming)
- [ ] Documentation Update 
- [ ] Other (explain)

## Further comments

- I have only tested the PCA plots on unit testing. This may affect other plots too
- I will be testing this fix also on other plots like UMap and some other plots maybe (need to check on which ones relate to Index issue).

### CODE REVIEW

File: OlinkAnalyze/tests/testthat/test-pca_plot.R [hunk 1]
```diff
@@ -218,6 +224,18 @@ test_that("minimal PCA plot", {
 
   vdiffr::expect_doppelganger("PCA plot - label outliers", pca_plot_outliers[[1]])
 
+  #Removing Index dependence in PCA plot
+  ids <- sample(npx_data1$SampleID, 100, replace = T)
+  ids <- gsub("A|B","",ids)
+  ids <- unique(sort(as.numeric(ids)))
```
Comment (AskPascal) [inline OlinkAnalyze/tests/testthat/test-pca_plot.R:230]: There is a control sample in the list of ids at this stage which is not numeric but is called "CONTROL_SMPLE_S" which leads to a warning here...

**Review by MasoumehSheikhi — DISMISSED** [review]
- [x] In "npxProcessing.R" file, remove Index from lines 56 and 160

- [x] Example: npx_data1 |> filter(!str_detect(SampleID,'CONT')) |> mutate(Index=if_else(Panel=='Olink Cardiometabolic', Index+1L, Index)) |>
     olink_pca_plot(quiet = TRUE)

**Comment by amrita-kar** [comment]
> * [x]  In "npxProcessing.R" file, remove Index from lines 56 and 160
> * [ ]  Example: npx_data1 |> filter(!str_detect(SampleID,'CONT')) |> mutate(Index=if_else(Panel=='Olink Cardiometabolic', Index+1L, Index)) |>
>   olink_pca_plot(quiet = TRUE)

**Comment by amrita-kar** [comment]
Also deleted 'Index' on line 157

**Comment by AskPascal** [comment]
As we now rely on SampleID being unique. Should we check for that somewhere? Or do we do that already?

Something along these lines might be helpful:
```
df %>% group_by(SampleID, OlinkID) %>% tally()
```

**Pull request merged**